src/components/InlineWidget/InlineWidget.tsx

```diff
@@ -37,14 +37,21 @@ class InlineWidget extends React.Component<Props> {
     this.destroyInlineWidget = this.destroyInlineWidget.bind(this);
   }
 
-  componentDidUpdate() {
-    this.destroyInlineWidget();
-    window.Calendly.initInlineWidget({
-      url: withPageSettings(this.props.url, this.props.pageSettings),
-      parentElement: this.widgetParentContainerRef.current!,
-      prefill: this.props.prefill,
-      utm: this.props.utm,
-    });
+  componentDidUpdate(prevProps: Props) {
```

Comment:
Awesome! Thanks for catching this!

It looks like using `React.PureComponent` will automatically implement `shouldComponentUpdate` by doing a shallow prop and state comparison - https://reactjs.org/docs/react-api.html#reactpurecomponent.

Could you try updating this class to extend `React.PureComponent` instead of `React.Component` to see if that resolves the issue without adding the `if` condition below?

Reply:
My only thought behind doing the manual check, is that it only checks the props that would actually require reloading the widget. For example, if the styles change. But if you'd prefer I can still make that change

Reply:
Didn't even think of that, great suggestion!